### DISCUSSION TITLE
50 ms offset limit

### DISCUSSION THREAD

**Discussion by AuracleTech** [ppy/osu#23824] [Q&A] [answered]
Since I restarted playing on linux I am getting about 25 ms faster response time than on win
If the map already had a bad offset on win then the latency saved by linux causes me to max out on the offset
This happened on multiple maps and my audio is not even tweaked for latency yet

https://github.com/ppy/osu/assets/40531871/70cd1f7a-168c-4379-8051-98f0ac27f843

**Comment by bdach** [comment]
https://github.com/ppy/osu/issues/21947#issuecomment-1368050734 is relevant. There is a Windows-exclusive 15ms offset, which may end up being applied to other platforms too and changing the situation here.

**Reply by AuracleTech** [reply]
In my opinion the default global offset should be set to 15 for windows by default, this is it's purpose after all.

**Reply by peppy** [reply]
If that's what you took away from reading the thread then I have questions. The conclusion was that it needs to be applied for linux as well, not that it should be moved to the UO display.

Regardless, what you're seeing is very audio playback high latency, and you're trying to adjust beyond the 50ms limitation (by making objects appear earlier). This is not normal. If you're seeing this on every beatmap simply adjust your universal offset in settings (not per-map offset). Hopefully changing that will be enough, but if not, you may need to fix your system configuration. We can't support helping linux users with such things because it's a minefield of custom configurations so you're on your own here.

If it's just the one beatmap, then please link the beatmap so we can investigate further.

**Reply by AuracleTech** [reply]
On windows the beatmaps offset were ~ -7 but on linux they are ~ +35
Many maps are +-30 away from the average, not all of them fortunately
I estimate around 1/75 ranked maps and 1/25 unranked maps
This is fine for most people, lots of players don't even realise there's latency &| never used the offset anyway
But even on windows I've had issues where +50 wasn't enough for some maps, it was just very rare (1/400 maybe)
On linux though it's a necessity to have +-100 on the slider like on stable

Tomorrow I will test some maps, half in my favourites (known local offset) and half randoms and check the offset I get from both
It'll give me a good idea of the state of offsets for the game

**Reply by peppy** [reply, marked as answer]
- Ranked maps with more than 10-15ms errors in the map are generally UNRANKED and fixed, so I highly doubt your claims
- The global offset allows adjustment +/-500. As I said above, please *use this instead*

![osu! 2023-06-09 at 02 31 58](https://github.com/ppy/osu/assets/191335/ea5027bf-7d70-4e99-8df5-b5a97755db1f)

If the above doesn't work for you, it's an issue with your system configuration or hardware. You'll need to figure this out.

**Reply by AuracleTech** [reply]
Ill start using it
